mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,462 @@
+    :param conda_env: {{ conda_env }}
+    :param code_paths: A list of local filesystem paths to Python file dependencies (or directories
+                       containing file dependencies). These files are *prepended* to the system
+                       path when the model is loaded.
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output
+                      :py:class:`Schema <mlflow.types.Schema>`.
+
+    :param input_example: {{ input_example }}
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :param model_config:
+        A dict of valid overrides that can be applied to a flow instance during inference.
+        These arguments are used exclusively for the case of loading the model as a ``pyfunc``
+        Model.
+        These values are not applied to a returned flow from a call to
+        ``mlflow.promptflow.load_model()``.
+        To override configs for a loaded flow with promptflow flavor,
+        please update the ``pf_model.context`` directly.
+
+
+        Configs that can be overridden includes:
+        ``connection.provider`` - The connection provider to use for the flow. Reach
+        https://microsoft.github.io/promptflow/how-to-guides/set-global-configs.html#connection-provider
+        for more details on how to set connection provider.
+
+
+        An example of providing overrides for a model to use azure machine
+        learning workspace connection:
+
+        .. code-block:: python
+
+            flow_folder = Path(__file__).parent / "basic"
+            flow = load_flow(flow_folder)
+
+            workspace_resource_id = (
+                "azureml://subscriptions/{your-subscription}/resourceGroups/{your-resourcegroup}"
+                "/providers/Microsoft.MachineLearningServices/workspaces/{your-workspace}"
+            )
+            model_config = {"connection.provider": workspace_resource_id}
+
+            with mlflow.start_run():
+                logged_model = mlflow.promptflow.log_model(
+                    flow, artifact_path="promptflow_model", model_config=model_config
+                )
+    """
+    import promptflow
+    from promptflow._sdk._mlflow import (
+        DAG_FILE_NAME,
+        Flow,
+        _merge_local_code_and_additional_includes,
+        remove_additional_includes,
+    )
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    # check if path exists
+    path = os.path.abspath(path)
```

Comment:
Feels like it would be clearer when path exists, user will see `Path 'C:/users/xx..../target'` already exists, rather than 'target' already exists, which is a little vague.

Reply:
Makes a lot sense! We may add this to the utility itself later, thanks.